lambda/github_usage.py

```diff
@@ -42,28 +43,34 @@ def process_message(message):
         "log_org_repo_team_members": audit.log_org_repo_team_members,
     }
     action = message["action"]
+    assert isinstance(action, str)
     process_action = actions[action]
     success = process_action(message)
     if not success:
         LOG.error("Processing failed for %s", action)
     return success
 
 
-def register(notification):
+def register(message: Dict[str, Any]) -> bool:
     """
     Register user.
 
     Record user token in SSM.
     Report to Splunk.
     """
-    username = notification["username"]
-    token = notification["user_secret"]
+    username = message["username"]
+    assert isinstance(username, str)
+
+    token = message["user_secret"]
+    assert isinstance(token, str)
+
```

Comment:
I like the use of asserts here, it's a good runtime contract for the function. But if I may suggest:

```python
assert str(username)
```
That way it'll blow up if the username can't be stringified, which is almost certainly what we want, rather than it be exactly a string and nothing else

Reply:
A lot of these `isinstance` checks are left over from when I was marking dicts with the correct types all the way through. Mypy can't type check nested dicts though even with `isinstance` type hints so I gave in any used `Dict[str, Any]` everywhere instead. I'm going to remove all these runtime checks to improve readability as they don't help the static type checking.

Reply:
That makes a lot of sense. I like to think of mypy as doing shallow type checking. "is this variable a dictionary" not "what is the shape of this dictionary".
I guess native classes would get you what you want, in an unsatisfactory way. 
I also wish we could just throw data around and have it be typechecked, but ah well!

Reply:
```assert str(username)```
Mypy doesn't recognise this as a type check and complains.